packages/ui-components/src/components/JsonViewer/JsonViewer.component.tsx

```diff
@@ -0,0 +1,606 @@
+import * as themes from "./themes"
+import { SearchInput } from "../SearchInput"
+
+// DEFAULT THEME (DARK)
+const DEFAULT_THEME: themes.JsonViewerTheme = {
+  base00: "var(--color-syntax-highlight-base00)", // background
+  base01: "var(--color-syntax-highlight-base01)", // toolbar: border, background
+  base02: "var(--color-syntax-highlight-base02)", // border, type background, border
+  base03: "var(--color-syntax-highlight-base03)", // -
+  base04: "var(--color-syntax-highlight-base04)", // size
+  base05: "var(--color-syntax-highlight-base05)", // types: "undefined"
+  base06: "var(--color-syntax-highlight-base06)", // -
+  base07: "var(--color-syntax-highlight-base07)", // key, brace
+  base08: "var(--color-syntax-highlight-base08)", // types: "NaN"
+  base09: "var(--color-syntax-highlight-base09)", // ..., types: "string"
+  base0A: "var(--color-syntax-highlight-base0A)", // types: "null", "regex"
+  base0B: "var(--color-syntax-highlight-base0B)", // types: "float"
+  base0C: "var(--color-syntax-highlight-base0C)", // array index
+  base0D: "var(--color-syntax-highlight-base0D)", // expanded icon, types: "date", "function"
+  base0E: "var(--color-syntax-highlight-base0E)", // collapsed icon, types: "boolean"
+  base0F: "var(--color-syntax-highlight-base0F)", // copy icon, types: "integer"
+}
+
+// indent size in pixel
+const INDENTATION_SIZE = 5
+// default truncate size
+const DEFAULT_TRUNCATE_LENGTH = 100
+// default indent width
+const DEFAULT_INDENT_WIDTH = 4
+// default expand setting
+const DEFAULT_EXPANDED = 1
+
+// map of color keys to theme
+const colorMap = (theme: themes.JsonViewerTheme): ThemeColors => ({
+  background: theme.base00,
+  ellipsis: theme.base09,
+  brace: theme.base07,
+  key: theme.base07,
+  index: theme.base0C,
+  size: theme.base04,
+  border: theme.base02,
+  highlight: {
+    foreground: theme.base06,
+    background: theme.base02,
+  },
+  toolbar: {
+    border: theme.base01,
+    background: theme.base01,
+    foreground: theme.base07,
+  },
+  icon: {
+    expanded: theme.base0D,
+    collapsed: theme.base0E,
+    expandAll: theme.base0E,
+  },
+  dataType: {
+    boolean: theme.base0E,
+    date: theme.base0D,
+    float: theme.base0B,
+    function: theme.base0D,
+    integer: theme.base0F,
+    string: theme.base09,
+    nan: theme.base08,
+    null: theme.base0A,
+    undefined: theme.base05,
+    regexp: theme.base0A,
+    background: theme.base02,
+  },
+})
+
+// get type of value
+const type = (value: unknown): TypeValueLabelType => {
+  if (value === null) return "null"
+  if (Array.isArray(value)) return "array"
+  if (value instanceof RegExp) return "regexp"
+  if (value instanceof Date) return "date"
+  const t = (typeof value).toLowerCase()
+  if (t === "number") {
+    if (Number.isNaN(value)) return "nan"
+    return Number.isInteger(value) ? "integer" : "float"
+  } else return t as TypeValueLabelType
+}
+
+interface ThemeColors {
```

Comment:
```

type HighlightColors = {
  foreground?: string
  background?: string
}

type ToolbarColors = {
  border?: string
} & HighlightColors

type IconColors = {
  expanded?: string
  collapsed?: string
  expandAll?: string
}

type DataTypeColors = {
  array?: string
  object?: string
  boolean?: string
  date?: string
  float?: string
  function?: string
  integer?: string
  string?: string
  nan?: string
  null?: string
  undefined?: string
  regexp?: string
  background?: string
}

interface ThemeColors {
  background?: string
  ellipsis?: string
  brace?: string
  key?: string
  index?: string
  size?: string
  border?: string
  highlight: HighlightColors
  toolbar: ToolbarColors
  icon: IconColors
  dataType: DataTypeColors
}

```

Reply:
I would modify it this way to split the big type on chunks and also remove some duplciations that way